src/lib/common/Kconfig

```diff
@@ -313,6 +332,16 @@ menuconfig HAVE_PIN_MUX
 
           If unsure, say Y.
 
+config MAINLOOP_FD
```

Comment:
If these are options, the build will break unless every single part of the code that depends on fd watches get properly left out if this is disabled. Too much ado over a oft unused feature.

Instead, I suggest enabling them conditionally using the C pre-processor in the `sol-common-buildopts.h.in` file:

    #if defined(SOL_PLATFORM_LINUX)
    #define SOL_MAINLOOP_FD_ENABLED 1
    #define SOL_MAINLOOP_FORK_WATCH 1
    #endif

That'll still enable code to be built depending on features being available on a platform or not.

I'd go even further and define `SOL_PLATFORM_POSIX` if `SOL_PLATFORM_LINUX` is enabled, and define these if that's the case. This way, if, say, a FreeBSD port comes along, one just defines `SOL_PLATFORM_POSIX`  if `SOL_PLATFORM_FREEBSD` and things should just work.